fabric-renderer-api-v1/src/main/java/net/fabricmc/fabric/mixin/renderer/client/MixinWeightedBakedModel.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.mixin.renderer.client;
+
+import java.util.List;
+import java.util.Random;
+import java.util.function.Supplier;
+
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.Unique;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import net.minecraft.block.BlockState;
+import net.minecraft.client.render.model.WeightedBakedModel;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.collection.WeightedPicker;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.BlockRenderView;
+
+import net.fabricmc.fabric.api.renderer.v1.model.FabricBakedModel;
+import net.fabricmc.fabric.api.renderer.v1.render.RenderContext;
+
+@Mixin(WeightedBakedModel.class)
+public class MixinWeightedBakedModel implements FabricBakedModel {
+	@Shadow
+	@Final
+	private int totalWeight;
+	@SuppressWarnings("rawtypes")
+	@Shadow
+	@Final
+	private List models; // WeightedBakedModel.Entry is not visible
```

Comment:
I would personally make the generic type in the list the accessor

Reply:
Is that legal and a good idea?

Reply:
Per type erasure, yes. Means you don't have to cast in that case.

Reply:
Attempted to do this
```
        @Shadow
        @Final
	private int totalWeight;
	@SuppressWarnings("rawtypes")
	@Shadow
	@Final
	private List<WeightedBakedModelEntryAccessor> models;
	@Unique
	boolean isVanilla = true;

	@SuppressWarnings("rawtypes")
	@Inject(at = @At("RETURN"), method = "<init>")
	private void onInit(List models, CallbackInfo cb) {
		for (int i = 0; i < models.size(); i++) {
			if (!((FabricBakedModel) models.get(i).getModel()).isVanillaAdapter()) {
				isVanilla = false;
				break;
			}
		}
	}
```
causes the compile error:
```
~/Programming/fabric/fabric-renderer-api-v1/src/main/java/net/fabricmc/fabric/mixin/renderer/client/MixinWeightedBakedModel.java:57: error: cannot find symbol
                        if (!((FabricBakedModel) models.get(i).getModel()).isVanillaAdapter()) {
                                                              ^
  symbol:   method getModel()
  location: class Object
```
additionally for the WeightedPicker.getAt calls I need to cast to `List` anyways

Reply:
Welp best not to make more of a mess then.

Reply:
Yo, you need to change
```java
private void onInit(List models, CallbackInfo cb) {
```
to
```java
private void onInit(List<WeightedBakedModelEntryAccessor> models, CallbackInfo cb) {
```
this will work cuz mixin doesn't check about generics when injecting

Reply:
@coolmineman needa update this field shadow to be parameterized too (use the accessor as type argument)